### DISCUSSION TITLE
Test suite

### DISCUSSION THREAD

**Discussion by jesperpedersen** [Hyperfoil/Horreum#375] [Ideas]
We need to define tasks for the test suite such it will run for all platforms.

Currently, I see two major areas that needs to be changed:

1) Integration tests
2) Dependency on OpenSSL 1.1.x

For 1) - recorded under #324 - there are assumptions about container naming is done. This works for `docker` based setups, but not for `podman` based ones. Should we try and use `GenericContainer` as the base and explicit set the container name as part of that ? We need to have integration tests - especially towards dependencies like KeyCloak - but are there other ideas we could make use off ?

For 2) - recorded under #300 - basically boils to using old algorithms in the existing web stack. I have created a Google Summer of Code project to look at the entire web stack and upgrade it. Since there is an existing work-around for OpenSSL 3.x based installation this may not as 1) for now.

Feel free to share ideas on this subject !

**Comment by johnaohara** [comment]
RE 1) I started i have a WIP in improving Integration tests: https://github.com/johnaohara/Horreum/commits/clearTestTimeout

This involves moving to `GenericContainer` but also using the keycloak client to configure the keycloak instance rather than using scripts.  The WIP also resolves issues with resource lifecycle management and aligns the Integration tests to the quarkus test lifecycle, allowing quarkus to manage the reosources,  instead of using junit lifecycle hooks. This will also allow us to ustilize qusrkus dev services in the future and remove the 'infra' directory completely.  'infra' scripts are broken on any platform apart from linux atm

**Reply by jesperpedersen** [reply]
I think that sounds like an excellent idea !

It would be ideal to support N+1 KeyCloak releases, but we will have to try it out. It is also a matter of what KeyCloak versions are deployed in the wild.

For PostgreSQL - we probably be a bit more selective in moving to a new version. Having the minimum requirement of 12 is good, and will work for quite a while.

Looking forward to the pull request !
